app/controllers/ems_common.rb

```diff
@@ -427,27 +427,66 @@ def update_button_reset
 
   def update_button_validate
     verify_ems = find_by_id_filtered(model, params[:id])
-    validate_credentials verify_ems
+    queue_credential_validation_task verify_ems
   end
   private :update_button_validate
 
-  def validate_credentials(verify_ems)
+  def queue_credential_validation_task(verify_ems)
     set_record_vars(verify_ems, :validate)
+    creds = set_record_creds(verify_ems)
+
     @in_a_form = true
     @changed = session[:changed]
 
     # validate button should say "revalidate" if the form is unchanged
     revalidating = !edit_changed?
-    result, details = verify_ems.authentication_check(params[:type], :save => revalidating)
-    if result
-      add_flash(_("Credential validation was successful"))
+
+    args = {
+      :ems        => verify_ems,
+      :creds      => creds,
```

Comment:
cc @blomquisg ^^  Definitely not for this PR, but let's try to clean up all this garbage and make the interface much simpler.  Also, I'm expecting that once the UI no longer needs the EMS, then `update_authentication` becomes significantly simpler because the whole `:save => false` craziness goes away.